zfs/zfs.go

```diff
@@ -683,6 +652,76 @@ func (a ZFSSendArgsUnvalidated) Validate(ctx context.Context) (v ZFSSendArgsVali
 	}, nil
 }
 
+func (a ZFSSendArgsUnvalidated) buildCommonSendArgs() []string {
+
+	args := make([]string, 0)
+
+	if a.Raw {
+		args = append(args, "-w")
+	}
+
+	if a.Properties {
+		args = append(args, "-p")
+	}
+
+	if a.BackupProperties {
+		args = append(args, "-b")
+	}
+
+	if a.LargeBlocks {
+		args = append(args, "-L")
+	}
+
+	if a.Compressed {
+		args = append(args, "-c")
+	}
+
+	if a.EmbeddedData {
+		args = append(args, "-e")
+	}
+
+	return args
+}
+
+// ResumeToken is allowed to be empty
+// a must already be validated
+//
+// SECURITY SENSITIVE because Encrypted must be handled correctly
+func (a ZFSSendArgsValidated) buildSendArgs() ([]string, error) {
+	args := a.buildCommonSendArgs()
+
+	// ResumeToken takes precedence, we assume that it has been validated
+	// to reflect what is described by the other fields
+	if a.ResumeToken != "" {
+		args = append(args, "-t", a.ResumeToken)
+		return args, nil
+	}
+
+	if a.Encrypted.B && !a.Raw {
+		args = append(args, "-w")
```

Comment:
why do we need this here? `ZFSSendArgsUnvalidated.buildCommonSendArgs` should already be setting this, right?

Reply:
No, buildCommon works on an unvalidated sendargs struct, and thus only checks raw, not encrypted. But we want to avoid adding `-w` for raw and then again for encrypted, if both are set, which I agree now should error during validation.

TL;DR: yes, we need it here, for the case Raw=false,Encrypted=true.

Reply:
>  buildCommon works on an unvalidated sendargs struct, and thus only checks raw, not encrypted

How about only ZFSSendArgsValidated can have abuildCommonSendArgs then?